lib/util/arr_fill_empty.js

```diff
@@ -0,0 +1,15 @@
+'use strict'
+
+const arrFillEmpty = (arr, fill = null) => {
```

Comment:
Wont be simpler to do this with a map function?

Reply:
With map we would have to adjust the function to return a value which in that case would use more memory. In current approach we check the reference and simply would modify existing object instead of created a new copy/clone of object